shared/utils/logic.ts

```diff
@@ -0,0 +1,336 @@
+import { AllowMyInfoBase, FieldBase, FieldResponse } from '../types'
+
+import {
+  FormCondition,
+  FormDto,
+  PreventSubmitLogic,
+  FormLogic,
+  LogicType,
+  ShowFieldLogic,
+  LogicConditionState,
+} from '../types/form'
+
+export type PickLogicSubset<T extends FormDto = FormDto> = Pick<
+  T,
+  | '_id'
+  | 'form_logics'
+  | 'form_fields'
+  | 'responseMode'
+  | 'form_logics'
+  | 'form_fields'
+>
```

Comment:
Should remove the duplicate props that are picked :P

Reply:
oh sorry didn't realize i duplicated them 😓